x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/service/ElasticServiceAccountsTests.java

```diff
@@ -379,6 +379,24 @@ public void testElasticEnterpriseSearchServerAccount() {
             assertThat(role.indices().allowedIndicesMatcher(RefreshAction.NAME).test(enterpriseSearchIndex), is(true));
             assertThat(role.indices().allowedIndicesMatcher("indices:foo").test(enterpriseSearchIndex), is(false));
         });
+
+        // Connector secrets. Enterprise Search has read and write access.
+        assertThat(role.cluster().check("cluster:admin/xpack/connector/secret/get", request, authentication), is(true));
+        assertThat(role.cluster().check("cluster:admin/xpack/connector/secret/post", request, authentication), is(true));
+
+        final IndexAbstraction dotConnectorSecretsIndex = mockIndexAbstraction(".connector-secrets" + randomAlphaOfLengthBetween(1, 20));
+        assertThat(role.indices().allowedIndicesMatcher(TransportDeleteAction.NAME).test(dotConnectorSecretsIndex), is(true));
```

Comment:
We don't need these checks, as this user won't be used to access the connector secrets, but the XPack internal user. We can add these tests [here](https://github.com/elastic/elasticsearch/blob/770671667eaad7eca8404785c0422bb54a4213f5/x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/security/user/InternalUsersTests.java#L71).